man/kernel-install.xml

```diff
@@ -264,6 +264,14 @@
             <para>Some other layout not understood natively by <command>kernel-install</command>.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term>auto</term>
+          <listitem>
+            <para>Pick layout automatically.  This inspect the kernel image and <filename>$BOOT</filename>
+            and will choose a layout based on what it finds.  Leaving layout blank has the same effect.  This
```

Comment:
```suggestion
            and will choose a layout based on what it finds. Leaving <varname>layout</varname> blank has the same effect. This
```
This should probably also say, that `uki` is chosen if the passed in stuff is a UKI, `bls` if `$BOOT/loader/entries.srel` or an entry token is found and `other` otherwise.

Reply:
The logic is actually more complex, and I have my doubts that it makes much sense to replicate that in the documentation.

Reply:
Only slightly so, no? The content of the srel file is checked to be `bls`. I'm on the side that one should be able to know what's about to happen without having to read the code, but I agree that it can be problematic since documentation and implementation already diverged.